cmd/root.go

```diff
@@ -27,11 +28,21 @@ var (
 	projectDirFlag string
 	configFileFlag string
 	assetsFlag     []string
+
+	// backcompatAsset is initialized once when the plugin starts
+	backcompatAsset *backcompatvalidator.BackCompatAsset
 )
 
 var rootCmd = &cobra.Command{
 	Use:   "conjure-plugin",
 	Short: "Run conjure-go based on project configuration",
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		// Discover and validate backcompat assets once when the plugin starts
+		// This ensures we fail fast if there are any issues with asset configuration
+		var err error
+		backcompatAsset, err = backcompatvalidator.New(configFileFlag, assetsFlag)
+		return err
+	},
```

Comment:
We should centralize asset loading -- it's strange that we're loading the backcompat asset here but the not the extensions provider one.

I see that the extension provider one loads assets in a manner more similar to the pervious one (https://github.com/palantir/godel-conjure-plugin/pull/543/files#diff-4f86eeac15e9163f72c4352e944d63335803c3fb2e61192a5f56cc9115d7d546R51-R56) -- I missed this since I wasn't a reviewer on that PR.

I can try to pull up an example, but I think it will be better if we centralize the asset loading code so that all the assets for the plugin follow the same pattern. I'll see if I can make a PR to refactor the existing state, after which this PR can just be updated to load its assets in an analogous manner.